pandas/tests/arrays/test_datetimelike.py

```diff
@@ -340,7 +340,7 @@ def test_from_array_keeps_base(self):
         arr = np.array(['2000-01-01', '2000-01-02'], dtype='M8[ns]')
         dta = DatetimeArray(arr)
 
-        assert dta._data is arr
+        assert dta._data.base is arr
```

Comment:
why this change?

Reply:
because a view of `arr` is taken, so the assertion is untrue as-is.

Reply:
then why was it passing on master? Because you changed it to take a view? Is it important to do that?

Reply:
Because I changed it back to use _from_sequence, which uses sequence_to_dt64ns, which yes takes a view.  And it is important in as much as in the PR implementing sequence_to_dt64ns when the original version only took a view if it was necessary, there was a request to do it unconditionally to save a line